### PR TITLE
agent: Use systemctl set-property for transient changes

### LINKED ISSUE #32: Use systemd drop-ins for temporary cgroup configuration
Today the agent is directly configuring a cgroup, i.e.:

https://github.com/openshift-virtualization/wasp-agent/blob/c6bb6dfc39563bab6c0fb3304bdfb2a8fc0a5385/contrib/agent.sh#L19

or

https://github.com/openshift-virtualization/wasp-agent/blob/c6bb6dfc39563bab6c0fb3304bdfb2a8fc0a5385/contrib/agent.sh#L38

However, we can move to using systemd drop-ins in order to align better to how croi and systemd interact.
To also avoid that systemd might reconcile - overwrite - our configuration.

https://www.flatcar.org/docs/latest/setup/systemd/drop-in-units/

### PR DESCRIPTION
**Pull request by fabiand**
Get away fr bash based _set

Fixes #32

### CODE REVIEW

**Comment by fabiand** [comment]
DO NOT MERGE yet

**Comment by Barakmor1** [comment]
I'm getting:

```
[vagrant@node01 ~]$ systemctl set-property --runtime system.slice MemorySwapMax=0 IODeviceLatencyTargetSec=$(findmnt / --output MAJ:MIN -n | sed "s/:.*/:0/") 50ms
Failed to parse IODeviceLatencyTargetSec value 252:0.
```

**Comment by fabiand** [comment]
Try

    $ systemctl set-property --runtime system.slice MemorySwapMax=0 IODeviceLatencyTargetSec="$(findmnt / -no SOURCE) 50ms"

File: contrib/agent.sh [hunk 1]
```diff
@@ -3,29 +3,28 @@
...
+  [[ -n "$SWAPPINESS" ]] && ( set -x ; sysctl -w vm.swappiness=$SWAPPINESS ; )
 
+  #
   echo "Tuning system.slice"
-
-  # Disable swap for system.slice
-  _set 0 $FSROOT/sys/fs/cgroup/system.slice/memory.swap.max
-
-  # Set latency target to protect the root slice from io trash
-  MAJMIN=$(findmnt $FSROOT/ --output MAJ:MIN -n | sed "s/:.*/:0/")  # fixme can be manually provided
```
Comment (Barakmor1) [inline contrib/agent.sh:17]: why did we change the minor to 0?

**Comment by Barakmor1** [comment]
> Try
> 
> ```
> $ systemctl set-property --runtime system.slice \
>   MemorySwapMax=0 \
>   IODeviceLatencyTargetSec="$(findmnt / -no SOURCE) 50ms"
> ```
> 
> https://www.freedesktop.org/software/systemd/man/latest/systemd.resource-control.html

There seems to be a parsing issue:
```
[core@virt-barak-416-sbbnk-worker-0-p2t86 ~]$ SOURCE=$(findmnt / -no SOURCE) && sudo systemctl set-property --runtime system.slice MemorySwapMax=0 IODeviceLatencyTargetSec=${SOURCE} 50m
Failed to parse IODeviceLatencyTargetSec value /dev/vda4[/ostree/deploy/rhcos/deploy/1c6b3c4414bec11bb664544780ba15c61ffb9bc7c2702b23ba3b87a404ea5cc9.1].
```
@fabiand @enp0s3 
Do you think we could use the MAJMIN approach with systemctl?

**Comment by fabiand** [comment]
@Barakmor1 the bug is that we are looking at the wrong path

In the agent it is $FSROOT - speak you need to look at the root filesystem.

**Pull request closed without merging**